src/pdn/src/PdnGen.tcl

```diff
@@ -3565,11 +3570,16 @@ proc generate_stripes {tag net_name} {
         set domain_xMax [$rect xMax]
         set domain_yMax [$rect yMax]
         # Do not create duplicate stripes if the voltage domain has the same pwr/gnd nets as the core domain
-        if {($net_name == [get_voltage_domain_power $domain_name] && $net_name != [get_voltage_domain_power [dict get $design_data core_domain]]) ||
+        #if {($net_name == [get_voltage_domain_power $domain_name] && $net_name != [get_voltage_domain_power [dict get $design_data core_domain]]) ||
+        #      ($net_name == [get_voltage_domain_ground $domain_name] && $net_name != [get_voltage_domain_ground [dict get $design_data core_domain]])} {}
+        
+        if {([lsearch [get_voltage_domain_power $domain_name] $net_name] != -1 && $net_name != [get_voltage_domain_power [dict get $design_data core_domain]]) ||
               ($net_name == [get_voltage_domain_ground $domain_name] && $net_name != [get_voltage_domain_ground [dict get $design_data core_domain]])} {
           set rail_width [get_rails_max_width]
           set area [list $domain_xMin [expr $domain_yMin - $rail_width / 2] $domain_xMax [expr $domain_yMax + $rail_width / 2]]
-          set area [adjust_area_for_core_rings $lay $area]
+          #Calculate the ring number of power_domain
+          set ring_number [expr [llength [get_voltage_domain_power $domain_name]] + [llength [get_voltage_domain_ground $domain_name]]]
+          set area [adjust_area_for_core_rings $lay $area $ring_number]
```

Comment:
Calculating ring number insider adjust_area_for_core_rings will prevent code repetition

Reply:
Same as the above. https://github.com/The-OpenROAD-Project/OpenROAD/pull/1118#discussion_r716957254